[6.3] Fix rocprof-sys-sample cannot get GPU related tracing

adjordje-amd opened the pull request:
After unloading the RCCL library, the library destructor triggers undefined behavior by invoking OnLoad function. 
Disable the RCCL backend to avoid issues with the library destructor and undefined behavior.

dgaliffiAMD commented:
Thanks, Aleks. 
This does fix the original issue for the transpose example;  `rocprof-sys-sample --PTDH -I all -- ./transpose` now includes "rocm-smi" samples. However, it seems to have introduced a new type failure for certain workloads. For example, when I run `rocprof-sys-sample -PTDH -I all -v 2 -- ls -al`, the trace fails to exit. We get stuck while tearing down:

```
[rocprof-sys][597260] Removing temporary file '/tmp/rocprofsys-ls-output/281783/sampling-597260.dat'...
[rocprof-sys][597260][0][rocprofsys_finalize] Finalized: 0.045335 sec wall_clock,    2.304 MB peak_rss,    2.359 MB page_rss, 0.010000 sec cpu_clock,   22.0 % cpu_util
```

adjordje-amd committed:
Disable RCCL backend to avoid issue with the RCCL

The RCCL library destructor triggers undefined behaviour by invoking OnLoad function.
Disabling the RCCL backend to prevent this issue.

dgaliffiAMD reviewed (approved):
Workaround for the 6.3 branch

dgaliffiAMD commented:
Rebasing change to the HEAD of `release/rocm-rel-6.3` branch

dgaliffiAMD committed:
Update VERSION

dgaliffiAMD commented:
I have updated the "patch" version, since this is the first change on `release/rocm-rel-6.3` since the last release.
